app/src/main/java/org/wikipedia/main/MainActivity.kt

```diff
@@ -0,0 +1,189 @@
+package org.wikipedia.main
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import android.view.View
+import androidx.appcompat.view.ActionMode
+import androidx.appcompat.widget.Toolbar
+import androidx.fragment.app.Fragment
+import org.wikipedia.Constants
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.activity.SingleFragmentActivity
+import org.wikipedia.appshortcuts.AppShortcuts.Companion.setShortcuts
+import org.wikipedia.databinding.ActivityMainBinding
+import org.wikipedia.navtab.NavTab
+import org.wikipedia.onboarding.InitialOnboardingActivity
+import org.wikipedia.page.PageActivity
+import org.wikipedia.page.tabs.TabActivity
+import org.wikipedia.settings.Prefs
+import org.wikipedia.suggestededits.SuggestedEditsTasksFragment
+import org.wikipedia.util.DimenUtil.dpToPx
+import org.wikipedia.util.DimenUtil.getDimension
+import org.wikipedia.util.FeedbackUtil.setButtonLongPressToast
+import org.wikipedia.util.FeedbackUtil.showTooltip
+import org.wikipedia.util.ResourceUtil.getThemedColor
+import org.wikipedia.views.TabCountsView
+
+class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callback {
+    private lateinit var binding: ActivityMainBinding
+
+    private var tabCountsView: TabCountsView? = null
+    private var controlNavTabInFragment = false
+    private var showTabCountsAnimation = false
+
+    override fun inflateAndSetContentView() {
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setShortcuts(this)
+        setImageZoomHelper()
+        if (Prefs.isInitialOnboardingEnabled() && savedInstanceState == null) {
+            // Updating preference so the search multilingual tooltip
+            // is not shown again for first time users
+            Prefs.setMultilingualSearchTutorialEnabled(false)
+
+            // Use startActivityForResult to avoid preload the Feed contents before finishing the initial onboarding.
+            // The ACTIVITY_REQUEST_INITIAL_ONBOARDING has not been used in any onActivityResult
+            startActivityForResult(InitialOnboardingActivity.newIntent(this), Constants.ACTIVITY_REQUEST_INITIAL_ONBOARDING)
+        }
+        setNavigationBarColor(getThemedColor(this, R.attr.nav_tab_background_color))
+        setSupportActionBar(binding.mainToolbar)
+        supportActionBar?.title = ""
+        supportActionBar?.setDisplayHomeAsUpEnabled(false)
+        binding.mainToolbar.navigationIcon = null
+    }
+
+    override fun onResume() {
+        super.onResume()
+        supportInvalidateOptionsMenu()
```

Comment:
Since it is deprecated, can we switch it to `invalidateOptionsMenu`?

Reply:
correct